examples/networking/ping_reply.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python
+"""
+ping_reply.py
+    - Hooks at Traffic Control (ingress queue) in Kernel using eBPF
+    - If packet is an ICMP echo (ping)
+        * Intercepts the packet at traffic control ingress queue
+        * Modifies ICMP packet type to ICMP reply
+        * Swaps src/dst ip and mac and updates chksum
+        * Redirects packet to the same interface as egress
+
+USAGE: sudo python ping_reply.py
+"""
+
+from bcc import BPF
+from pyroute2 import IPRoute, IPDB
+import sys
+
+prog = """
+  #include <bcc/proto.h>
+  #include <uapi/linux/if_ether.h>
+  #include <uapi/linux/in.h>
+  #include <uapi/linux/icmp.h>
+ 
+  #define IP_SRC_OFF 26
```

Comment:
The use of static offsets and u8 *cursor at the same time is mixing two different flavors. I would try to use one approach or the other. If you go with cursor (preferred), then you can do something like 'incr_cksum_l3(&ip->hchecksum, orig_icmp_type, new_icmp_type);', which would support dynamic parsing of the ip header in case of different ethertypes.